presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java

```diff
@@ -42,33 +43,66 @@
 import static com.facebook.presto.spi.StandardErrorCode.COMPILER_ERROR;
 import static com.facebook.presto.sql.gen.BytecodeUtils.invoke;
 import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
 
 public class ExpressionCompiler
 {
     private final Metadata metadata;
 
-    private final LoadingCache<CacheKey, Class<? extends PageProcessor>> pageProcessors = CacheBuilder.newBuilder().maximumSize(1000).build(
-            new CacheLoader<CacheKey, Class<? extends PageProcessor>>()
+    private final LoadingCache<ProcessorCacheKey, Class<? extends PageProcessor>> pageProcessors = CacheBuilder.newBuilder().maximumSize(1000).build(
+            new CacheLoader<ProcessorCacheKey, Class<? extends PageProcessor>>()
             {
                 @Override
-                public Class<? extends PageProcessor> load(CacheKey key)
+                public Class<? extends PageProcessor> load(ProcessorCacheKey key)
                         throws Exception
                 {
                     return compile(key.getFilter(), key.getProjections(), new PageProcessorCompiler(metadata), PageProcessor.class);
                 }
             });
 
-    private final LoadingCache<CacheKey, Class<? extends CursorProcessor>> cursorProcessors = CacheBuilder.newBuilder().maximumSize(1000).build(
-            new CacheLoader<CacheKey, Class<? extends CursorProcessor>>()
+    private final LoadingCache<ProcessorCacheKey, Class<? extends CursorProcessor>> cursorProcessors = CacheBuilder.newBuilder().maximumSize(1000).build(
+            new CacheLoader<ProcessorCacheKey, Class<? extends CursorProcessor>>()
             {
                 @Override
-                public Class<? extends CursorProcessor> load(CacheKey key)
+                public Class<? extends CursorProcessor> load(ProcessorCacheKey key)
                         throws Exception
                 {
                     return compile(key.getFilter(), key.getProjections(), new CursorProcessorCompiler(metadata), CursorProcessor.class);
                 }
             });
 
+    private final LoadingCache<JoinFilterCacheKey, Class<? extends JoinFilterFunction>> joinFilterFunctions = CacheBuilder.newBuilder().maximumSize(1000).build(
+            new CacheLoader<JoinFilterCacheKey, Class<? extends JoinFilterFunction>>()
+            {
+                @Override
+                public Class<? extends JoinFilterFunction> load(JoinFilterCacheKey key)
+                        throws Exception
+                {
+                    ClassDefinition classDefinition = new ClassDefinition(
```

Comment:
Move this code to a method, say `compileJoinFilterInternal`, so the cache field declaration is small (like the ones above)